transactions/repository.go

```diff
@@ -0,0 +1,78 @@
+package transactions
+
+import (
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/storage"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/satori/go.uuid"
+)
+
+const (
+	// ErrTransactionMissing error when transaction doesn't exist in Repository.
+	ErrTransactionMissing = errors.Error("transaction doesn't exist")
+
+	// ErrKeyConstructionFailed error when the key construction failed.
+	ErrKeyConstructionFailed = errors.Error("failed to construct transaction key")
+)
+
+// Repository can be implemented by a type that handles storage for transactions.
+type Repository interface {
+	Get(identity common.Address, id uuid.UUID) (*Transaction, error)
+	Save(transaction *Transaction) error
```

Comment:
The db supports only a `Put(key,value)`.  we build a `create` and `update` on the lowest level around it.
Here we build a new `put(key, value)` around on a higher level.

Reply:
Just to make things easier, We dont need to support create and update specifically here since we will using it to update logs